index.rst

```diff
@@ -1,59 +1,367 @@
+In many cases, write access will actually be prohibited (see :ref:`access-controls`).
+
+When migrations are necessary due to changes in the repository format (something that is *always* preceded by an RFC with explicit CCB approval), a new ``repo_<YYYYMMDD>`` directory and database/schema pair will be created, and files will shared via hard links until/unless the old repository is retired.
+
+.. note::
+
+   **TODO**: Are hard links viable here from a sysadmin/GPFS perspective?
+   They certainly would make things easier, because they'd let each repository have its own file without actual duplicationof storage.  Note that Gen3 treats datasets as completely atomic and immutable (aside from deletion), so there is no chance of one repository updating another unexpectedly via hard links.
+
+We will also designate two other non-repository subdirectories of ``/datasets`` for specific roles:
+
+ - ``/datasets/external`` holds files produced by other projects or surveys that may be of use to multiple users but does not fit into the LSST data model.  This includes the original versions of reference catalogs (e.g. Gaia DR2), truth catalogs (e.g. from DESC DC2), dust maps, etc.  Each subdirectory should have a descriptive README, and no files should be put in ``datasets/external`` itself.
+
+ - ``/datasets/testing`` holds git LFS repositories that are used in CI and rarely change.  These are provided for convenience, and should be updated when their git master branches are; there will be no attempt to make old versions available.
+
+Finally, we propose that all raw ingestion into the shared repository be done with symbolic links or in-place (outside-root) ingests, pointing to read-only filesystems - the existing ``/lsstdata`` for Rubin Observatory data, and a new ``/external-raw`` filesystem for raw data from other instruments.
+
+.. note ::
+
+   **TODO**: This document should probably say more about staging locations for (at least) raws that haven't been ingested yet, for both Rubin Observatory instruments and precursor instruments (which I imagine might be quite different in this respect).  I'd like to at least be confident that the filesystem locations and permissions I'm proposing are consistent with how data arrives and gets ingested.  I don't know enough to write that up myself.
+
+.. _access-controls:
+
+Access Controls
+===============
+
+The current Gen3 registry architecture does not allow any fine-grained access control in the repository database; we instead rely on "friendly users" being careful and respectful of this shared space.
+
+At the same time, we will use filesystem access controls to protect shared and per-user files, and we plan to implement some checks in the Butler client itself to make it at least extremely difficult to *accidentally* cause problems.
```

Comment:
How should we handle users who want to make their user space `collections` unreadable to anyone else, or grant permission to specific people to access some but not all of their `collections`?

Reply:
We are not going to support that for our developers at NCSA. Everyone will be able to see everyone else's collections.

In the science platform we will have a client/server registry where the server will know your user ID and your group membership.